File: src/test/java/com/orange/dojo/elephantcarpaccio/terminal/TerminalTest.java
package com.orange.dojo.elephantcarpaccio.terminal;

import static org.fest.assertions.Assertions.assertThat;

import java.io.PrintWriter;
import java.io.StringReader;
import java.io.StringWriter;
import java.util.Scanner;

import org.junit.Test;

public class TerminalTest {

  @Test
  public void I_Can_Enter_My_Name_From_The_Terminal() {
    // Given
    String expectedName = "Jeff";

    Scanner input = new Scanner(new StringReader(expectedName));
    PrintWriter output = new PrintWriter(new StringWriter());

    Terminal terminal = new Terminal(input, output);

    // When
    String name = terminal.whatIsYourName();

    // Then
    assertThat(name).isEqualTo(expectedName);
  }
  
  @Test
  public void I_Can_Enter_My_Age_From_The_Terminal() {
    // Given
    int expectedAge = 35;
    Scanner input = new Scanner(new StringReader(String.format("%d", expectedAge)));
    PrintWriter output = new PrintWriter(new StringWriter());
    
    Terminal terminal = new Terminal(input, output);
    
    // When
    int age = terminal.whatIsYourAge();
    
    // Then
    assertThat(age).isEqualTo(expectedAge);
  }
}

File: kotlin/src/test/kotlin/com/orange/dojo/elephantcarpaccio/terminal/TerminalTest.kt
package com.orange.dojo.elephantcarpaccio.terminal

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.io.PrintWriter
import java.io.StringReader
import java.io.StringWriter
import java.util.*

class TerminalTest {

    @Test
    fun `I can enter my name from the terminal`() {
        // Given
        val expectedName = "Jeff"

        val input = Scanner(StringReader(expectedName))
        val output = PrintWriter(StringWriter())

        val terminal = Terminal(input, output)

        // When
        val name = terminal.whatIsYourName()

        // Then
        assertThat(name).isEqualTo(expectedName)
    }

    @Test
    fun `I can enter my age from the terminal`() {
        // Given
        val expectedAge = 35
        val input = Scanner(StringReader(String.format("%d", expectedAge)))
        val output = PrintWriter(StringWriter())

        val terminal = Terminal(input, output)

        // When
        val age = terminal.whatIsYourAge()

        // Then
        assertThat(age).isEqualTo(expectedAge)
    }
}
